monai/utils/img2tensorboardutils.py

```diff
@@ -0,0 +1,118 @@
+    summary_image_str = summary_pb2.Summary.Image(
+        height=10, width=10, colorspace=1, encoded_image_string=img_str)
+    image_summary = summary_pb2.Summary.Value(tag=tag, image=summary_image_str)
+    return summary_pb2.Summary(value=[image_summary])
+
+
+def make_animated_gif_summary(tag,
+                              tensor,
+                              max_out=3,
+                              animation_axes=(1,),
+                              image_axes=(2, 3),
+                              other_indices=None,
+                              scale_factor=1):
+    """
+    Creates an animated gif out of an image tensor and returns Summary.
+
+    Args:
+        tag: Data identifier
+        tensor: tensor for the image, expected to be in CDHW format
+        max_out: maximum number of slices to animate through
+        animation_axes: axis to animate on (not currently used)
+        image_axes: axes of image (not currently used)
+        other_indices: (not currently used)
+        scale_factor: amount to multiply values by (if the image data is between 0 and 1, using 255 for this value will
+        scale it to displayable range)
+    """
+
+    if max_out == 1:
+        suffix = '/image'
+    else:
+        suffix = '/image/{}'
+    if other_indices is None:
+        other_indices = {}
+    axis_order = [0] + animation_axes + image_axes
+
+    slicing = []
+    for i in range(len(tensor.shape)):
+        if i in axis_order:
+            slicing.append(slice(None))
+        else:
+            other_ind = other_indices.get(i, 0)
+            slicing.append(slice(other_ind, other_ind + 1))
+    tensor = tensor[tuple(slicing)]
+
+    for it_i in range(min(max_out, list(tensor.shape)[0])):
+        inp = [
+            tag + suffix.format(it_i), tensor[it_i, :, :, :]
+        ]
+        summary_op = _image3_animated_gif(inp, scale_factor)
+    return summary_op
+
+
+def add_animated_gif(writer, tag, image_tensor, max_out, scale_factor, global_step=None):
+    """
+    Creates an animated gif out of an image tensor and writes it with SummaryWriter.
+
+    Args:
+        writer: Tensorboard SummaryWriter to write to
+        tag: Data identifier
+        image_tensor: tensor for the image to add, expected to be in CDHW format
+        max_out: maximum number of slices to animate through
+        scale_factor: amount to multiply values by (if the image data is between 0 and 1, using 255 for this value will
+        scale it to displayable range)
+        global_step: Global step value to record
+    """
+    writer._get_file_writer().add_summary(make_animated_gif_summary(tag, image_tensor, max_out=max_out,
+                                                                    animation_axes=[1], image_axes=[2, 3],
+                                                                    scale_factor=scale_factor), global_step)
+
+
+def add_animated_gif_no_channels(writer, tag, image_tensor, max_out, scale_factor, global_step=None):
```

Comment:
If this GIF feature only works for CDHW data, why not **combine** add_animated_gif() and add_animated_gif_no_channels()?

I think:
option 1: you can get the shape of input data and add 1 channel dim if not existing,
option 2: force the input to be CDHW and ask user to convert their data to be CDHW.
What do you think?

Reply:
I haven't actually made the decision to only support CDHW. In fact, I would like to incorporate batch with a parameter to specify how many items of the batch to load in a later version.

Reply:
Hi @nvkevlu ,

OK, thanks for your explanation.
But why you need to show 2D image in **GIF** format? So I thought this tool only works for 3D/CDHW.
Thanks.

Reply:
2D won't be supported but CDHW and NDHW may be.

Reply:
Hi @nvkevlu ,

At least for this MVP version MONAI, we finalized to only support CDHW, no need to consider other shape formats.
Thanks.